source/routing/journey_pattern_container.h

```diff
@@ -0,0 +1,143 @@
+/* Copyright © 2001-2015, Canal TP and/or its affiliates. All rights reserved.
+
+This file is part of Navitia,
+    the software to build cool stuff with public transport.
+
+Hope you'll enjoy and contribute to this project,
+    powered by Canal TP (www.canaltp.fr).
+Help us simplify mobility and open public transport:
+    a non ending quest to the responsive locomotion way of traveling!
+
+LICENCE: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+Stay tuned using
+twitter @navitia
+IRC #navitia on freenode
+https://groups.google.com/d/forum/navitia
+www.navitia.io
+*/
+
+#pragma once
+
+#include "raptor_utils.h"
+
+namespace navitia { namespace type {
+
+struct PT_Data;
+struct VehicleJourney;
+struct DiscreteVehicleJourney;
+struct FrequencyVehicleJourney;
+
+}}// namespace navitia::type
+
+namespace navitia { namespace routing {
+
+struct JourneyPattern;
+struct JourneyPatternPoint;
+using JppIdxT = Idx<JourneyPatternPoint>;
+using JpIdxT = Idx<JourneyPattern>;
+
+// TODO: constructor private with JourneyPatternContainer friend?
+struct JourneyPatternPoint {
+    SpIdx sp_idx;
+    uint32_t order;
+    inline bool operator==(const JourneyPatternPoint& other) const {
+        return sp_idx == other.sp_idx && order == other.order;
+    }
+};
+
+// TODO: constructor private with JourneyPatternContainer friend?
+struct JourneyPattern {
+    std::vector<JppIdxT> jpps;
+    std::vector<const navitia::type::DiscreteVehicleJourney*> discrete_vjs;
+    std::vector<const navitia::type::FrequencyVehicleJourney*> freq_vjs;
+
+    inline bool operator==(const JourneyPattern& other) const {
+        return jpps == other.jpps
+            && discrete_vjs == other.discrete_vjs
+            && freq_vjs == other.freq_vjs;
+    }
+    template<typename VJ> std::vector<const VJ*>& get_vjs();
+};
+
+
+struct JourneyPatternContainer {
+    using Jps = std::vector<JourneyPattern>;
+    using Jpps = std::vector<JourneyPatternPoint>;
+    using jp_iterator = IdxMapIterator<JpIdxT, Jps::const_iterator>;
+    using jpp_iterator = IdxMapIterator<JppIdxT, Jpps::const_iterator>;
+    using jp_range = boost::iterator_range<jp_iterator>;
+    using jpp_range = boost::iterator_range<jpp_iterator>;
+
+    void init(const navitia::type::PT_Data&);
```

Comment:
No, I can't, dataRaptor is inited each time the data is updated, and we want to reuse the memory at each init